auth-foundation/src/main/java/com/okta/authfoundation/client/DeviceTokenCookieJar.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2023-Present Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.authfoundation.client
+
+import okhttp3.Cookie
+import okhttp3.CookieJar
+import okhttp3.HttpUrl
+
+class DeviceTokenCookieJar : CookieJar {
+    private val deviceTokenCookieBuilder = Cookie.Builder()
+        .name("DT")
+        .value(DeviceTokenProvider.deviceToken)
+        .secure()
+
+    override fun loadForRequest(url: HttpUrl): List<Cookie> {
+        val deviceTokenCookie = deviceTokenCookieBuilder.domain(url.host)
+            .build()
+        return listOf(deviceTokenCookie)
```

Comment:
Will this _only_ send the device token cookie? Or will it also include cookies previously set by the server? I think the implementation should be that it injects the DT cookie _in addition_ to all regular cookies managed by the device.

Reply:
I'm now returning all the cookies saved for the url domain